src/index.ts

```diff
@@ -449,18 +378,76 @@ async function main() {
             new Option("-i, --interactive", "Enable interactive interface")
         )
         .action(async (jobID, options) => {
-            await resetPosts(options.interactive, jobID);
+            await resetPosts(spinner, page, options.interactive, jobID);
         });
+}
 
-    loginCommand.description("Login and save credentials").action(async () => {
-        await login();
-    });
+function configureLoginCommand(command: Command, sso: SSO) {
+    return command
+        .command("login")
+        .description("Login and save credentials")
+        .action(async () => {
+            await sso.login();
+        });
+}
+
+function configureLogoutCommand(command: Command, sso: SSO) {
+    return command
+        .command("logout")
+        .description("Remove saved credentials")
+        .action(() => {
+            sso.logout();
+        });
+}
+
+function configureCommand(
+    command: Command,
+    page: Page,
+    spinner: Ora,
+    sso: SSO
+) {
+    command.description(
+        "Greenhouse is a command-line tool that provides helpers to automate " +
+            "interactions with the Canonical Greenhouse website."
+    );
 
-    logoutCommand.description("Remove saved credentials").action(() => {
-        logout();
+    const replicateCommand = configureReplicateCommand(command, page, spinner);
+    const deletePostsCommand = configureDeleteCommand(command, page, spinner);
+    const resetPostsCommand = configureResetCommand(command, page, spinner);
+    const loginCommand = configureLoginCommand(command, sso);
+    const logoutCommand = configureLogoutCommand(command, sso);
+
+    command.configureHelp({
+        visibleCommands: () => {
+            return [
+                replicateCommand,
+                resetPostsCommand,
+                loginCommand,
+                logoutCommand,
+            ];
+        },
     });
+}
 
-    await program.parseAsync(process.argv);
+async function main() {
+    setupSentry();
+    const spinner = ora();
+    let browser: Browser | null = null;
+    try {
+        const program = new Command();
+        const sso = new SSO(spinner);
```

Comment:
When doing the SSO first, some commands like `help` & `logout`, `sso` will be required for them where it's in reality not required at all for the command `help` for example.

Reply:
:+1: